test/framework/docker_logcollector.go

```diff
@@ -45,6 +46,22 @@ func machineContainerName(cluster, machine string) string {
 
 func (k DockerLogCollector) CollectMachineLog(ctx context.Context, managementClusterClient client.Client, m *clusterv1.Machine, outputPath string) error {
 	containerName := machineContainerName(m.Spec.ClusterName, m.Name)
+	return k.collectLogsFromNode(outputPath, containerName)
+}
+
+func (k DockerLogCollector) CollectMachinePoolLog(ctx context.Context, managementClusterClient client.Client, m *expv1.MachinePool, outputPath string) error {
+	var reterr error
+	for _, instance := range m.Status.NodeRefs {
+		containerName := machineContainerName(m.Spec.ClusterName, instance.Name)
+		if err := k.collectLogsFromNode(filepath.Join(outputPath, instance.Name), containerName); err != nil {
+			// collecting logs is best effort so we proceed to the next instance even if we encounter an error.
+			reterr = err
```

Comment:
what about using `kerrors.NewAggregate` etc here? This way we could catch all errors and not only the last one.